x-pack/legacy/plugins/apm/public/components/app/Settings/AgentConfigurations/List/ConfirmDeleteModal.tsx

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState } from 'react';
+import { EuiConfirmModal, EuiOverlayMask } from '@elastic/eui';
+import { NotificationsStart } from 'kibana/public';
+import { i18n } from '@kbn/i18n';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { AgentConfigurationListAPIResponse } from '../../../../../../../../../plugins/apm/server/lib/settings/agent_configuration/list_configurations';
+import { getOptionLabel } from '../../../../../../../../../plugins/apm/common/agent_configuration_constants';
+import { callApmApi } from '../../../../../services/rest/createCallApmApi';
+import { useApmPluginContext } from '../../../../../hooks/useApmPluginContext';
+
+type Config = AgentConfigurationListAPIResponse[0];
+
+interface Props {
+  config: Config;
+  onCancel: () => void;
+  onConfirm: () => void;
+}
+
+export function ConfirmDeleteModal({ config, onCancel, onConfirm }: Props) {
+  const [isDeleting, setIsDeleting] = useState(false);
+  const { toasts } = useApmPluginContext().core.notifications;
+
+  return (
+    <EuiOverlayMask>
+      <EuiConfirmModal
+        title={i18n.translate('xpack.apm.agentConfig.deleteModal.title', {
+          defaultMessage: `Delete configuration`
+        })}
+        onCancel={onCancel}
+        onConfirm={async () => {
+          setIsDeleting(true);
+          await deleteConfig(config, toasts);
+          setIsDeleting(false);
+          onConfirm();
+        }}
+        cancelButtonText={i18n.translate(
+          'xpack.apm.agentConfig.deleteModal.cancel',
+          { defaultMessage: `No, cancel` }
```

Comment:
This phrasing seems unnecessary. What not just "Cancel" and "Delete"? I'm guessing it's meant to be there to clarify but to me it might just cause more confusion.

Reply:
Will change.
On. that topic @formgeist, do we any any guidelines on dialog messages?

Reply:
EUI do have guidelines for modal messaging https://elastic.github.io/eui/#/guidelines/modals - I think in this case it should just say "Cancel".